File: app/src/main/java/io/inabsentia/superhangman/data/dao/IMatchDAO.java
package io.inabsentia.superhangman.data.dao;

import android.content.Context;

import java.util.List;

import io.inabsentia.superhangman.data.dto.MatchDTO;

public interface IMatchDAO {
    void add(MatchDTO dto) throws DALException;
    MatchDTO get(int id) throws DALException;
    List<MatchDTO> getAll() throws DALException;
    void update(MatchDTO dto) throws DALException;
    void remove(int id) throws DALException;
    void removeAll() throws DALException;
    void load(Context context) throws DALException;
    void save(Context context) throws DALException;
}

File: app/src/main/java/io/inabsentia/superhangman/data/dao/IMatchDAO.kt
package io.inabsentia.superhangman.data.dao

import android.content.Context

import io.inabsentia.superhangman.data.dto.MatchDTO

interface IMatchDAO {
    val all: List<MatchDTO>

    @Throws(DALException::class)
    fun add(dto: MatchDTO)

    @Throws(DALException::class)
    fun get(id: Int): MatchDTO

    @Throws(DALException::class)
    fun update(dto: MatchDTO)

    @Throws(DALException::class)
    fun remove(id: Int)

    @Throws(DALException::class)
    fun removeAll(context: Context)

    @Throws(DALException::class)
    fun load(context: Context)

    @Throws(DALException::class)
    fun save(context: Context)
}